docs/development/releasing.rst

```diff
@@ -1,233 +1,386 @@
 
-   Next, run towncrier and confirm that the fragments can be deleted::
+Additional notes
+----------------
 
-       towncrier --version 5.0
+Do not render the changelog with towncrier at this point. This should only be done just before the final
+release. However, it is up to the discretion of the release manager whether to
+open 'practice' pull requests to do this as part of the beta/release candidate
+process (but they should not be merged in) - if so the process for rendering the changelog is described
+in :ref:`render-changelog`.
 
-   Then add and commit those changes with::
+.. _release-procedure-tagging:
 
-      $ git add CHANGES.rst
-      $ git commit -m "Finalizing changelog for v<version>"
+Tagging the first release candidate
+-----------------------------------
 
-   Push to GitHub and open a pull request for merging this into the release branch,
-   e.g. v5.0.x.
+Assuming all the CI passes, you should now be ready to do a first release
+candidate! Ensure you have a GPG key pair available for when git needs to sign
+the tag you create for the release.  See :ref:`key-signing-info` for more on
+this.
 
-   In cases where an LTS branch and a different release branch are being maintained,
-   the changelog should be rendered on both branches separately, and only the
-   rendering from the non-LTS release branch should be forward-ported to main.
+Make sure your local release branch is up-to-date with the upstream release
+branch, then tag the latest commit with the ``-s`` option, including an ``rc1``
+suffix, e.g.::
 
-   .. note::
+      $ git tag -s v5.0rc1 -m "Tagging v5.0rc1"
 
-      We render the changelog on the latest release branch and forward-port it
-      rather than rendering on main and backporting, since the latter would
-      render all news fragments into the changelog rather than only the ones
-      intended for the e.g. v5.0.x release branch.
+Push up the tag to the `astropy core repository`_, e.g.::
 
-#. Once the changelog pull request is merged, update your release branch to
-   match the upstream version, then (on the release branch), tag the merge
-   commit for the changelog changes with ``v<version>``, being certain to sign
-   the tag with the ``-s`` option::
+      $ git push upstream v5.0rc1
 
-      $ git tag -s v<version> -m "Tagging v<version>"
+.. note::
+
+   It might be tempting to use the ``--tags`` argument to ``git push``,
+   but this should *not* be done, as it might push up some unintended tags.
+
+At this point if all goes well, the wheels and sdist will be build
+in the `Azure core package pipeline`_ and uploaded to PyPI!
 
-#. Push up the tag to the `astropy core repository`_::
+In the event there are any issues with the wheel building for the tag
+(which shouldn't really happen if it was passing for the release branch),
+you'll have to fix whatever the problem is. Make sure you delete the
+tag::
 
-      $ git push upstream v<tag version>
+   git tag -d v<version>
```

Comment:
the tag needs to be deleted remote, too, otherwise I believe git push would complain about conflict when trying to push the new one again